src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -348,11 +348,11 @@ protected ISelectionModel Selection
                     }
 
                     InitializeSelectionModel(_selection);
-
-                    if (_oldSelectedItems != SelectedItems)
+                    var seletcedItems = SelectedItems;
+                    if (_oldSelectedItems.TryGetTarget(out var oldSelectedItems) || oldSelectedItems != seletcedItems)
```

Comment:
This isn't equivalent to the comparison before: it will now always be true as long as we have an `oldSelectedItems`. 
```suggestion
_oldSelectedItems.TryGetTarget(out var oldSelectedItems);
if (oldSelectedItems != selectedItems)
```